Source/WebKit/WebProcess/WebPage/WebPage.cpp

```diff
@@ -7553,14 +7553,7 @@ RefPtr<Element> WebPage::elementForContext(const ElementContext& elementContext)
     if (elementContext.webPageIdentifier != m_identifier)
         return nullptr;
 
-    auto* document = Document::allDocumentsMap().get(elementContext.documentIdentifier);
-    if (!document)
-        return nullptr;
-
-    if (document->page() != m_page.get())
-        return nullptr;
-
-    return document->searchForElementByIdentifier(elementContext.elementIdentifier);
+    return Element::fromIdentifier(elementContext.elementIdentifier);
```

Comment:
This can now return an element that is no longer in the `WebPage` if it’s been moved to another document in another `WebPage` by JavaScript, for example. In the old code the connection would have been broken if the element was ever moved out of the document it started in.

I don’t know if that change in semantic is a problem. As long as we don’t care about the super-edge-case of moving out of a document and back in, we can easily restore the semantic by checking what document the element is in, and what frame that’s in, and what page that’s in. If we don’t need the document identifier since the element identifier is no longer scoped to a specific document, then we could consider removing it from `ElementContext`.